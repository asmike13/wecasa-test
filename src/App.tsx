import './App.css';
import ListContainer from 'src/components/ListContainer';
import { Provider } from 'react-redux';
import { store } from 'src/store';

function App() {
	return (
		<div className="App">
			<Provider store={store}>
				<ListContainer />
			</Provider>
		</div>
	);
}

export default App;
