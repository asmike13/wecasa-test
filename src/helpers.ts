export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
