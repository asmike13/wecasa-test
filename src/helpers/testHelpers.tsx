import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
