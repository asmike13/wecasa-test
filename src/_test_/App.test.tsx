import renderer from 'react-test-renderer';
import App from 'src/App'
import { StoreWrapper } from 'src/helpers/testHelpers';

it('should render app as snapshot', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<App />
		</StoreWrapper>,
	)
	
	expect(snapshot).toMatchSnapshot()
})
