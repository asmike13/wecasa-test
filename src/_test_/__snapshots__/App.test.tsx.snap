// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app as snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Wecasa Test
  </h1>
  <div>
    Loading ...
  </div>
  <div
    className="cart-margin"
    style={
      Object {
        "height": "0px",
      }
    }
  />
  <div
    className="cart"
  >
    Panier vide
  </div>
</div>
`;
